File: app/src/main/java/com/leodev/kiittimetable/Activities/zoomlinkActivity.java
package com.leodev.kiittimetable.Activities;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.text.util.Linkify;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Spinner;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.google.gson.Gson;
import com.leodev.kiittimetable.Adapters.ZoomAdapter;
import com.leodev.kiittimetable.R;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import static android.content.SharedPreferences.*;

public class zoomlinkActivity extends AppCompatActivity {
    private RecyclerView recyclerView;
    private RecyclerView.LayoutManager layoutManager;
    private List<String> list;
    private ZoomAdapter adapter;
    private Object Editor;

    private EditText linkEditText;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.actrivity_zoom_link);
        recyclerView=findViewById(R.id.recyclerview_zoom);
        layoutManager=new LinearLayoutManager(this);
        recyclerView.setLayoutManager(layoutManager);
        list= Arrays.asList(getResources().getStringArray(R.array.subjects));
        adapter=new ZoomAdapter(list);
        recyclerView.setHasFixedSize(true);
        recyclerView.setAdapter((RecyclerView.Adapter) adapter);


        SharedPreferences sharedPreferences=getSharedPreferences("Zoom Links",MODE_PRIVATE);
        SharedPreferences.Editor editor=sharedPreferences.edit();
        View rvItem;
        for(int i=0;i<recyclerView.getChildCount();i++){

        }
    }
}

File: app/src/main/java/com/leodev/kiittimetable/Activities/ZoomLinkActivity.kt
package com.leodev.kiittimetable.Activities

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.firestore.SetOptions
import com.google.firebase.firestore.ktx.firestore
import com.google.firebase.ktx.Firebase
import com.leodev.kiittimetable.Adapters.ZoomAdapter
import com.leodev.kiittimetable.R
import kotlinx.android.synthetic.main.activity_zoom_link.*

class ZoomLinkActivity : AppCompatActivity() {

    val db = Firebase.firestore
    val auth = FirebaseAuth.getInstance()
    val subjects = arrayListOf("Data Algorithm and Analysis",
        "Data Mining and Data Warehouse/Big Data",
        "Software Engineering",
        "Artificial Intelligence/Cryptography",
        "Computer Networking",
        "High Performance Computing")
    val linksArray: MutableMap<String, String?> = mutableMapOf()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_zoom_link)

        val zoomSharedPrefs = getSharedPreferences("zoom", Context.MODE_PRIVATE)
        for(sub in subjects){
            val sc = getSubjectCode(sub)
            val t = zoomSharedPrefs.getString(sc, null)
            linksArray[sc] = t
        }

        val adapter = ZoomAdapter(subjects, zoomSharedPrefs, linksArray)

        rv_zoom_links.adapter = adapter
        rv_zoom_links.layoutManager = LinearLayoutManager(this)

        btn_zoom_links.setOnClickListener {
            for(sub in subjects){
                val subjectCode = getSubjectCode(sub)
                val link = zoomSharedPrefs.getString(subjectCode, null)

                val uid = auth.currentUser?.uid!!
                val info = hashMapOf("link" to link)
                val timetableDB = db.collection("users").document(uid).collection("timetable")
                timetableDB.document(subjectCode).set(info, SetOptions.merge())
            }
            finish()
        }
    }

    private fun getSubjectCode(sub: String): String {
        return when(sub){
            "Data Algorithm and Analysis" -> "CSE301"
            "Computer Networking" -> "CSE302"
            "High Performance Computing" -> "CSE303"
            "Software Engineering" -> "CSE304"
            "Artificial Intelligence/Cryptography" -> "CSE305"
            "Data Mining and Data Warehouse/Big Data" -> "CSE306"
            else -> ""
        }
    }
}